network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractBranchImpl.java

```diff
@@ -605,32 +604,15 @@ public <E extends Extension<T>> Collection<E> getExtensions() {
     }
 
     @Override
-    public boolean connect() {
-        return getTerminal1().connect() && getTerminal2().connect();
-    }
-
-    @Override
-    public boolean connect(Predicate<Switch> isTypeSwitchToOperate) {
-        return getTerminal1().connect(isTypeSwitchToOperate) && getTerminal2().connect(isTypeSwitchToOperate);
-    }
-
-    @Override
-    public boolean connect(Predicate<Switch> isTypeSwitchToOperate, ThreeSides side) {
-        return getTerminal(side.toTwoSides()).connect(isTypeSwitchToOperate);
-    }
-
-    @Override
-    public boolean disconnect() {
-        return getTerminal1().disconnect() && getTerminal2().disconnect();
-    }
-
-    @Override
-    public boolean disconnect(Predicate<Switch> isSwitchOpenable) {
-        return getTerminal1().disconnect(isSwitchOpenable) && getTerminal2().disconnect(isSwitchOpenable);
-    }
-
-    @Override
-    public boolean disconnect(Predicate<Switch> isSwitchOpenable, ThreeSides side) {
-        return getTerminal(side.toTwoSides()).disconnect(isSwitchOpenable);
+    public List<Terminal> getTerminals(ThreeSides side) {
```

Comment:
before we got an error if SIDE_3 in toTwoSide() and now we just return the empty list and the remaining code does nothing ? Is it on purpose ?

Reply:
I tried to mimic how it works in Core, where we stream the terminals and filter by comparing sides (so for a branch, Side.Three would get an empty list)

Reply:
I can add a warning if you want

Reply:
it's more about changing the behavior without notice. It's better to be aligned with Core but needs to be described

Reply:
I added this in the breaking change section in case users were using the exception that was thrown